packages/component/src/Styles/useInjectCustomProperties.ts

```diff
@@ -0,0 +1,58 @@
+import { hooks } from 'botframework-webchat-api';
+import { makeCreateStyles } from 'botframework-webchat-styles';
```

Comment:
Should we have a new package?

Should we create a new private package for isolation, but bundle it in when compiling?

The other side to think... who can use `bf-wc-styles`? Can it be used without `bc-wc-component`?

Should we put all `styleOptions` and related stuff (say, rectification), inside the new `bf-wc-styles` package?

I like we say, `bf-wc-styles/react`. Good to use named import for different ecosystems.

Reply:
At the moment the `styles` package is internal. It is dedicated for providing necessary tools for applying styles. Think of it as `emotion`, but with smaller scope and tightened to our selection of tools.

Even in case we expose it, it seems we would need a better name for it (without `botframework-webchat` prefix) as while being our set of tools, it doesn't limit its usage in any way.

We can make it another package for styling Web Chat, but I don't think it's necessary at this point.

Reply:
In case we'll need a `component2` package which leverages `core`, `api` etc for new experience, we can think on increasing the scope for `styles` to cover things we want to share between the `component` and `component2`.

Reply:
I was thinking `styles` depends on `component`, thus I was thinking about it that way. 🤣

I think it makes sense to have a new package. If it lives outside of the `botframework-webchat` prefix, then we should have a different version number not tie to Web Chat.

Reply:
As it is internal for now, we can make this later. Our setup is pretty lightweight and may indeed be useful for making CSS modules to be available today.

This needs the following improvements to be useful for community:
- The build plugin should use [`unplugin`](https://unplugin.unjs.io/) so it supports most of bundlers, not only esbuild.
- We should probably switch from `style` elements to [`CSSStyleSheet`](https://web.dev/constructable-stylesheets/) for handling styles as this is something we expect to get from `import style from '*.css' with { type: 'css' };`
- We should improve `makeCreateStyles` and `createStyles` to make it less confusing at least

A few of these are going along with current Web Chat goals:
- We use only esbuild/tsup
- The `CSSStyleSheet` is not yet very useful for our needs: we have to convert back and forth to style elements.

Reply:
Different POV, but all are good stuff.